designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessingTypeDeployedScenariosProvider.scala

```diff
@@ -0,0 +1,27 @@
+package pl.touk.nussknacker.engine.api.deployment
+
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.deployment.DeploymentData
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ProcessingTypeDeployedScenariosProvider {
+
+  def getDeployedScenarios(implicit ec: ExecutionContext): Future[List[DeployedScenarioData]]
```

Comment:
What is the reasoning behind placing this in a separate class from DeploymentManager?

Reply:
`DeploymentManager` or `DeploymentService`? I'm not sure what are you asking about. This interface is used by DM, but delivered by DS.